pkg/test/framework/components/echo/echoboot/echoboot.go

```diff
@@ -15,12 +15,86 @@
 package echoboot
 
 import (
+	"istio.io/istio/pkg/test"
 	"istio.io/istio/pkg/test/framework/components/echo"
 	"istio.io/istio/pkg/test/framework/components/echo/kube"
 	"istio.io/istio/pkg/test/framework/resource"
 )
 
 // NewBuilder for Echo Instances.
-func NewBuilder(ctx resource.Context) echo.Builder {
-	return kube.NewBuilder(ctx)
+func NewBuilder(ctx resource.Context, clusters ...resource.Cluster) Builder {
+	return builder{
+		kubeBuilder: kube.NewBuilder(ctx),
+	}.WithClusters(clusters...)
+}
+
+// Builder is a superset of echo.Builder, which allows deploying the same echo configuration across clusters.
+type Builder interface {
```

Comment:
Why not just change the echo builder API?

Reply:
I was trying to avoid changing the kube implementation to include WithClusters